addons/test_mail/tests/test_mail_activity.py

```diff
@@ -215,6 +231,150 @@ def test_action_feedback_attachment(self):
+                'res_id': origin_2.id,
+                'date_deadline': today_utc + relativedelta(hours=8),
+                'user_id': self.user_utc.id,
+            })
+
+            origin_2_activity_2 = origin_2_activity_1.copy()
+            origin_2_activity_2.user_id = self.user_australia
+            origin_2_activity_3 = origin_2_activity_1.copy()
+            origin_2_activity_3.date_deadline -= relativedelta(hours=8)
+            origin_2_activity_4 = origin_2_activity_1.copy()
+            origin_2_activity_4.date_deadline = datetime(2020, 1, 2, 0, 0, 0)
+
+            self.assertEqual(origin_2_activity_1.state, 'planned')
+            self.assertEqual(origin_2_activity_2.state, 'today')
+            self.assertEqual(origin_2_activity_3.state, 'today')
+            self.assertEqual(origin_2_activity_4.state, 'planned')
+
+            all_partners = self.env['res.partner'].search([])
+
+            result = self.env['res.partner'].search([('activity_state', '=', 'today')])
+            self.assertTrue(len(result) > 0)
+            self.assertEqual(result, all_partners.filtered(lambda p: p.activity_state == 'today'))
+
+            result = self.env['res.partner'].search([('activity_state', 'in', ('today', 'overdue'))])
+            self.assertTrue(len(result) > 0)
+            self.assertEqual(result, all_partners.filtered(lambda p: p.activity_state in ('today', 'overdue')))
+
+            result = self.env['res.partner'].search([('activity_state', 'not in', ('today',))])
+            self.assertTrue(len(result) > 0)
+            self.assertEqual(result, all_partners.filtered(lambda p: p.activity_state not in ('today',)))
+
+            result = self.env['res.partner'].search([('activity_state', '=', False)])
+            self.assertTrue(len(result) > 10, "There is more than 10 partners without an activity schedule on it")
+            self.assertEqual(result, all_partners.filtered(lambda p: not p.activity_state))
+
+            result = self.env['res.partner'].search([('activity_state', 'not in', ('planned', 'overdue', 'today'))])
+            self.assertTrue(len(result) > 10, "There is more than 10 partners without an activity schedule on it")
+            self.assertEqual(result, all_partners.filtered(lambda p: not p.activity_state))
+
+            # test tricky case when the domain will be reversed in the search method
+            # because of falsy value
+            result = self.env['res.partner'].search([('activity_state', 'not in', ('today', False))])
+            self.assertTrue(len(result) > 0)
+            self.assertEqual(result, all_partners.filtered(lambda p: p.activity_state not in ('today', False)))
+
+            result = self.env['res.partner'].search([('activity_state', 'in', ('today', False))])
+            self.assertTrue(len(result) > 0)
+            self.assertEqual(result, all_partners.filtered(lambda p: p.activity_state in ('today', False)))
+
+    def test_mail_activity_mixin_search_state_2(self):
```

Comment:
_2 does not mean anything ... try to find a name on your method that explains what your test is about :)